azurerm/internal/services/containers/container_registry_resource.go

```diff
@@ -71,18 +71,41 @@ func resourceContainerRegistry() *schema.Resource {
 				Optional: true,
 				Default:  false,
 			},
-
+			// TODO 3.0 - Remove below property
 			"georeplication_locations": {
-				Type:     schema.TypeSet,
-				MinItems: 1,
-				Optional: true,
+				Type:          schema.TypeSet,
+				Optional:      true,
+				Deprecated:    "Deprecated in favour of `georeplications`",
+				Computed:      true,
+				ConflictsWith: []string{"georeplications"},
 				Elem: &schema.Schema{
 					Type:         schema.TypeString,
 					ValidateFunc: validation.StringIsNotEmpty,
 				},
 				Set: location.HashCode,
 			},
 
+			"georeplications": {
+				Type:          schema.TypeList,
```

Comment:
I think List is enough, since service always return the response in the same order as the request.

ref: https://github.com/microsoft/terraform-provider-azurerm/wiki/R-1006

Reply:
Another reason to use list, is that there's an issue about optional-computed set, state won't be updated after applying, unless refresh.
https://github.com/hashicorp/terraform-plugin-sdk/issues/618

Reply:
If order doesn't matter this should be a set

Reply:
I'll change it to set.